common/Resources/ti.internal/extensions/node/internal/util/inspect.js

```diff
@@ -0,0 +1,1695 @@
+				const sourcePattern = /(.*):(\d+):(\d+)/;
+				let symbolName = 'unknown';
+				if (atSymbolIndex !== -1) {
+					symbolName = stackLine.slice(0, atSymbolIndex);
+				}
+
+				const sourceMatch = source.match(sourcePattern);
+				if (sourceMatch) {
+					let filePath = sourceMatch[1];
+					const lineNumber = sourceMatch[2];
+					const column = sourceMatch[3];
+					if (filePath.startsWith('file:')) {
+						filePath = filePath.replace(`file://${Ti.Filesystem.resourcesDirectory}`, '');
+					}
+
+					return `\n    at ${symbolName} (${filePath}:${lineNumber}:${column})`;
+				} else {
+					return `\n    at ${symbolName} (${source})`;
+				}
+			}).join('');
+		}
+	}
+
+	// A stack trace may contain arbitrary data. Only manipulate the output
+	// for "regular errors" (errors that "look normal") for now.
+	const name = err.name || 'Error';
+	let len = name.length;
+	if (constructor === null
+			|| name.endsWith('Error')
+			&& stack.startsWith(name)
+			&& (stack.length === len || stack[len] === ':' || stack[len] === '\n')) {
+		let fallback = 'Error';
+		if (constructor === null) {
+			const start = stack.match(/^([A-Z][a-z_ A-Z0-9[\]()-]+)(?::|\n {4}at)/)
+				|| stack.match(/^([a-z_A-Z0-9-]*Error)$/);
+			fallback = start && start[1] || '';
+			len = fallback.length;
+			fallback = fallback || 'Error';
+		}
+		const prefix = getPrefix(constructor, tag, fallback).slice(0, -1);
+		if (name !== prefix) {
+			if (prefix.includes(name)) {
+				if (len === 0) {
+					stack = `${prefix}: ${stack}`;
+				} else {
+					stack = `${prefix}${stack.slice(len)}`;
+				}
+			} else {
+				stack = `${prefix} [${name}]${stack.slice(len)}`;
+			}
+		}
+	}
+
+	// Ignore the error message if it's contained in the stack.
+	let pos = err.message && stack.indexOf(err.message) || -1;
+	if (pos !== -1) {
+		pos += err.message.length;
+	}
+	// Wrap the error in brackets in case it has no stack trace.
+	let stackStart = stack.indexOf('\n    at', pos);
+
+	if (stackStart === -1) {
+		stack = `[${stack}]`;
+	} else if (ctx.colors) {
+		// Highlight userland code and node modules.
+		let newStack = stack.slice(0, stackStart);
+		const lines = stack.slice(stackStart + 1).split('\n');
+		for (const line of lines) {
+			// This adds underscores to all node_modules to quickly identify them.
+			let nodeModule;
+			newStack += '\n';
+			let pos = 0;
+			while (nodeModule = nodeModulesRegExp.exec(line)) {
+				// '/node_modules/'.length === 14
+				newStack += line.slice(pos, nodeModule.index + 14);
+				newStack += ctx.stylize(nodeModule[1], 'module');
+				pos = nodeModule.index + nodeModule[0].length;
+			}
+			newStack += pos === 0 ? line : line.slice(pos);
+		}
+		stack = newStack;
+	}
+	// The message and the stack have to be indented as well!
+	if (ctx.indentationLvl !== 0) {
+		const indentation = ' '.repeat(ctx.indentationLvl);
+		stack = stack.replace(/\n/g, `\n${indentation}`);
+	}
+	return stack;
+}
+
+function formatPromise(ctx, value, recurseTimes) {
```

Comment:
<!--
  0 failure: 
  2 warning:  common/Resources/..., common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/internal/util/inspect.js;
  Line: 1000;
-->

- :warning: common/Resources/ti.internal/extensions/node/internal/util/inspect.js line 1000 – 'recurseTimes' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)
- :warning: common/Resources/ti.internal/extensions/node/internal/util/inspect.js line 1000 – 'value' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)